### PR TITLE
Testable cmd init

### PR DESCRIPTION
**Pull request by freeekanayaka**
This branch consists of a couple of separate commits:

1) make cmdInit testable by extracting its global variables and "boundaries", so most of its logic can be run in isolation in a unit test. A sample unit test has been added as exemplification, I plan to add more as I add the logic for the new "pre-seed" feature as per #2834.

2) extract a few inline functions into methods, for easier reading of the core logic.

The commit messages of the individual commits provide a bit of extra narrative.

I'm bundling these two changes together for convenience, but I'm happy to split them in separate branches if more appropriate.

### CODE REVIEW

**Comment by freeekanayaka** [comment]
jenkins: test this please

File: lxd/main_test.go [hunk 1]
```diff
@@ -13,7 +14,8 @@ import (
 
 func mockStartDaemon() (*Daemon, error) {
 	d := &Daemon{
-		MockMode: true,
+		MockMode:  true,
+		tlsConfig: &tls.Config{},
```
Comment (stgraber) [inline lxd/main_test.go]: Why is that needed?

**Comment by freeekanayaka** [comment]
Stéphane Graber <notifications@github.com> writes:

> stgraber commented on this pull request.
>
>
>
>> @@ -13,7 +14,8 @@ import (
>  
>  func mockStartDaemon() (*Daemon, error) {
>  	d := &Daemon{
> -		MockMode: true,
> +		MockMode:  true,
> +		tlsConfig: &tls.Config{},
>
> Why is that needed?

It's needed because otherwise this line in api_1.0.go:

https://github.com/freeekanayaka/lxd/blob/testable-cmd-init/lxd/api_1.0.go#L163

would cause a panic. IIRC that line is triggered when the Go client
tries to connect (e.g. with lxd.ConnectLXDUnix()).

In the follow-up branch I'm making this a bit more polished by
generating a random certificate for the test suite:

```go
	certBytes, keyBytes, err := shared.GenerateMemCert(false)
	if err != nil {
		return nil, err
	}
	cert, err := tls.X509KeyPair(certBytes, keyBytes)
	if err != nil {
		return nil, err
	}

	d := &Daemon{
		MockMode: true,
		tlsConfig: &tls.Config{
			Certificates: []tls.Certificate{cert},
		},
	}
```
